tests/unit/browser/test_adblock.py

```diff
@@ -0,0 +1,434 @@
+
+BLOCKLIST_HOSTS = ('localhost',
+                   'mediumhost.io',
+                   'malware.badhost.org',
+                   '4-verybadhost.com',
+                   'ads.worsthostever.net')
+
+CLEAN_HOSTS = ('goodhost.gov', 'verygoodhost.com')
+
+URLS_TO_CHECK = ('http://localhost',
+                 'http://mediumhost.io',
+                 'http://malware.badhost.org',
+                 'http://4-verybadhost.com',
+                 'http://ads.worsthostever.net',
+                 'http://goodhost.gov',
+                 'ftp://verygoodhost.com'
+                 'http://qutebrowser.org')
+
+
+@pytest.fixture
+def data_tmpdir(monkeypatch, tmpdir):
+
+    """Set tmpdir as datadir"""
+
+    tmpdir = str(tmpdir)
+    monkeypatch.setattr('qutebrowser.utils.standarddir.data', lambda: tmpdir)
+
+
+# XXX Why does read_hosts needs basedir to be None
+# in order to print message 'run adblock-update to read host blocklist' ?
+# browser/adblock.py line 138
+# TODO Replace with unittest mock.mock / See test webelem
+class BaseDirStub:
+
+    """Mock for objreg.get('args') called in adblock.HostBlocker.read_hosts."""
+
+    def __init__(self):
+        self.basedir = None
+
+
+@pytest.yield_fixture
+def basedir():
+
+    """Register a Fake basedir."""
+
+    args = BaseDirStub()
+    objreg.register('args', args)
+    yield args
+    objreg.delete('args')
+
+
+class FakeDownloadItem(QObject):
+
+    """Mock browser.downloads.DownloadItem."""
+
+    finished = pyqtSignal()
+
+    def __init__(self, fileobj, name):
+        super().__init__()
+        self.fileobj = fileobj
+        self.name = name
+        self.successful = True
+
+
+class FakeDownloadManager:
+
+    """Mock browser.downloads.DownloadManager."""
+
+    def get(self, url, fileobj, **kwargs):
+
+        """Return a FakeDownloadItem instance with a fileobj
+           whose content is copied from the file the given url links to.
+        """
+
+        download_item = FakeDownloadItem(fileobj, name=url.path())
+        with open(url.path(), 'rb') as fake_url_file:
+            # Ensure cursors are at position 0 before copying
+            fake_url_file.seek(0)
+            download_item.fileobj.seek(0)
+            shutil.copyfileobj(fake_url_file, download_item.fileobj)
+        return download_item
+
+
+@pytest.yield_fixture
+def download_stub(win_registry):
+
+    """Register a FakeDownloadManager."""
+
+    stub = FakeDownloadManager()
+    objreg.register('download-manager', stub,
+                    scope='window', window='last-focused')
+    yield stub
+    objreg.delete('download-manager', scope='window', window='last-focused')
+
+
+def create_zipfile(directory, files, zipname='test'):
+
+    """Return a path to a created zip file
+
+       Args :
+       - directory : path object where to create the zip file
+       - files : list of paths to each file to add in the zipfile
+       - zipname : name to give to the zip file.
+    """
+
+    directory = str(directory)
```

Comment:
Why not use the [py.path](http://pylib.readthedocs.org/en/latest/path.html) object which makes those things easier instead of converting it to a string immediately?

Something like this (untested):

```python
zipfile_path = directory / zipname + '.zip'
with zipfile.ZipFile(str(zipfile_path), 'w') as new_zipfile:
    ...
return str(zipfile_path)
```